nexus/internal/filetransfer/file_transfer_manager.go

```diff
@@ -202,13 +204,25 @@ func (fm *FileTransferManager) transfer(task *Task) error {
 	case UploadTask:
 		err = fm.fileTransfer.Upload(task)
 	case DownloadTask:
+		if err := fm.ensureDownloadRootDir(task.Path); err != nil {
```

Comment:
Nit: I don't think this should be the file transfer manager's responsibility, let's move it to the callsite. Also, this way we will only check it once instead of separately for each file.

Reply:
I think the sdk team moved it here: https://github.com/wandb/wandb/blob/56670ebbebaa03700574d88ce945818557a9227e/nexus/internal/filetransfer/file_transfer_default.go#L82-L92